packages/wdio-cli/src/launcher.js

```diff
@@ -292,11 +297,12 @@ class Launcher {
             caps,
             specs,
             server,
-            execArgv
+            execArgv,
+            retries
         })
         worker.on('message', ::this.interface.onMessage)
         worker.on('error', ::this.interface.onMessage)
-        worker.on('exit', ::this.endHandler)
+        worker.on('exit', ({ cid, exitCode }) => this.endHandler({ cid, exitCode, specs, retries }))
```

Comment:
The worker has access to `specs` and `retries` and can pass them along in the exit event so that we can keep `worker.on('exit', ::this.endHandler)` here